Formula/glib.rb

```diff
@@ -19,11 +19,12 @@ class Glib < Formula
   depends_on "meson" => :build
   depends_on "ninja" => :build
   depends_on "pkg-config" => :build
-  depends_on "python@3.10" => :build
   depends_on "gettext"
-  depends_on "libffi"
   depends_on "pcre"
 
+  uses_from_macos "libffi", since: :catalina
+  uses_from_macos "python"
```

Comment:
This does mean utilities won't work out-of-the-box on Mojave and older since they need Python 3.

This is reason we originally didn't choose this approach as `uses_from_macos "python", since: :catalina` can trigger audit failure

Reply:
> This does mean utilities won't work out-of-the-box on Mojave and older since they need Python 3, which could cascade into breaking dependent builds.

Mojave support is "best-effort" --
I don't think supporting Mojave can justify the hacks in `glib-utils`.

> This is reason we originally didn't choose this approach as `uses_from_macos "python", since: :catalina` can trigger audit failure

What audit failure was that?

Reply:
The ones seen when mixing `node` (which uses alias for Python3) and `emscripten` (#95866)

Reply:
If I had to guess, `HOMEBREW_SIMULATE_MACOS_ON_LINUX` is taking code paths of older version of macOS.

Reply:
That sounds right. We probably don't want that. CC @Rylan12 in case there's an easy fix.

Reply:
`HOMEBREW_SIMULATE_MACOS_ON_LINUX` simulates the oldest allowable version of macOS[^1]. This was a decision made before the API project so I can't fully comment on the original reasoning. I think the idea is that all `uses_from_macos` deps will be treated as `uses_from_macos` regardless of a `since:` argument.

Reply:
What do you think about making this the oldest _supported_ release of macOS? That's almost always what we're interested in testing for in CI anyway.

Reply:
https://github.com/Homebrew/brew/pull/13714

Reply:
> What do you think about making this the oldest _supported_ release of macOS?

Off the top of my head, I think that would be fine. Looks like there wasn't really much discussion about it when it was added, so probably fine.

CC: @Bo98

Reply:
The philosophy behind choosing the oldest macOS is that it includes the most dependencies (99.9% of the time). For auditing reasons this is usually a good thing.

We would have the same issue on Linux if the conflict audit wasn't disabled there. I would not rely on this remaining the case - I've been wanting to change that for a long time now.

The conflict audit is really a special case. I'd say the way forward is to special case the `python` alias to not trigger the conflict audit, since those formulae are typically not bound to a particular version of Python. That's the real issue I feel.

Reply:
I've merged https://github.com/Homebrew/brew/pull/13714. Now we only need https://github.com/Homebrew/homebrew-core/pull/108343.

Reply:
```suggestion
  uses_from_macos "python", since: :catalina
```

Reply:
Yep. This needs another run for the self-hosted runner and to rebuild some dependents against system `libffi` anyway.